File: 2996-smallest-missing-integer-greater-than-sequential-prefix-sum/2996-smallest-missing-integer-greater-than-sequential-prefix-sum.cpp
class Solution {
public:
    int missingInteger(vector<int>& nums) {
        int countSum = nums[0];
        for(int i =1; i<nums.size();i++){
            if(nums[i-1]+1 == nums[i]) countSum += nums[i];
            else break;
        }
        sort(nums.begin(),nums.end());
        for(int i = 0; i< nums.size(); i++){
            if(countSum == nums[i]){
                countSum++;
            }
        }
        return countSum;
        
    }
};

File: 2996-smallest-missing-integer-greater-than-sequential-prefix-sum/2996-smallest-missing-integer-greater-than-sequential-prefix-sum.java
class Solution {
    public int missingInteger(int[] nums) {
        int countSum = nums[0];
        for (int i = 1; i < nums.length; i++) {
            if (nums[i - 1] + 1 == nums[i]) countSum += nums[i];
            else break;
        }
        Arrays.sort(nums);
        for (int i = 0; i < nums.length; i++) {
            if (countSum == nums[i]) {
                countSum++;
            }
        }
        return countSum;
    }
}
